### ISSUE TITLE
Add `INTEGER` to as_result_metadata

### ISSUE THREAD

**Issue by criccomini** [tekumara/fakesnow#16]
On 0.4.1, I create this table:

```sql
CREATE TABLE IF NOT EXISTS test_types (
    TEST_BIGINT BIGINT,
    TEST_INTEGER INTEGER,
    TEST_SMALLINT SMALLINT,
    TEST_FLOAT FLOAT,
    TEST_FLOAT8 FLOAT8,
    TEST_FLOAT4 FLOAT4,
    TEST_DOUBLE FLOAT,
    TEST_DOUBLE_PRECISION DOUBLE PRECISION,
    TEST_REAL REAL,
    TEST_BOOLEAN BOOLEAN,
    TEST_VARCHAR VARCHAR(100),
    TEST_STRING STRING,
    TEST_TEXT TEXT,
    TEST_CHAR CHAR(10),
    TEST_NVARCHAR NVARCHAR(100),
    TEST_NVARCHAR2 NVARCHAR(100),
    TEST_CHAR_VARYING CHAR VARYING(100),
    TEST_NCHAR_VARYING NCHAR VARYING(100),
    TEST_NCHAR NCHAR(100),
    TEST_CHARACTER CHARACTER(100),
    TEST_BINARY BINARY,
    TEST_VARBINARY VARBINARY,
    TEST_BLOB BLOB,
    TEST_DATE DATE,
    TEST_TIMESTAMP TIMESTAMP,
    TEST_DATETIME DATETIME,
    TEST_TIME TIME,
    TEST_DECIMAL DECIMAL(10,2),
    TEST_NUMERIC NUMERIC(10,2),
    TEST_NUMBER NUMBER(10,2),
    TEST_TINYINT TINYINT,
    TEST_BYTEINT BYTEINT
);
```

I then open a cursor and call `cursor.description`. The call fails with:

```
recap/readers/dbapi.py:29: in to_recap
    names = [name[0].upper() for name in cursor.description]
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:77: in description
    with self._duck_conn.cursor() as cur:
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:84: in description
    meta = FakeSnowflakeCursor._describe_as_result_metadata(cur.fetchall())  # noqa: SLF001
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:297: in _describe_as_result_metadata
    meta = [
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:298: in <listcomp>
    as_result_metadata(column_name, column_type, null)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

column_name = 'ordinal_position', column_type = 'INTEGER', _ = 'YES'

    def as_result_metadata(column_name: str, column_type: str, _: str) -> ResultMetadata:
        # see https://docs.snowflake.com/en/user-guide/python-connector-api.html#type-codes
        # and https://arrow.apache.org/docs/python/api/datatypes.html#type-checking
        if column_type == "BIGINT":
            return ResultMetadata(
                name=column_name, type_code=0, display_size=None, internal_size=None, precision=38, scale=0, is_nullable=True               # noqa: E501
            )
        elif column_type.startswith("DECIMAL"):
            match = re.search(r'\((\d+),(\d+)\)', column_type)
            if match:
                precision = int(match[1])
                scale = int(match[2])
            else:
                precision = scale = None
            return ResultMetadata(
                name=column_name, type_code=0, display_size=None, internal_size=None, precision=precision, scale=scale, is_nullable=True    # noqa: E501
            )
        elif column_type == "VARCHAR":
            # TODO: fetch internal_size from varchar size
            return ResultMetadata(
                name=column_name, type_code=2, display_size=None, internal_size=16777216, precision=None, scale=None, is_nullable=True      # noqa: E501
            )
        elif column_type == "DOUBLE":
            return ResultMetadata(
                name=column_name, type_code=1, display_size=None, internal_size=None, precision=None, scale=None, is_nullable=True          # noqa: E501
            )
        elif column_type == "BOOLEAN":
            return ResultMetadata(
                name=column_name, type_code=13, display_size=None, internal_size=None, precision=None, scale=None, is_nullable=True         # noqa: E501
            )
        elif column_type == "DATE":
            return ResultMetadata(
                name=column_name, type_code=3, display_size=None, internal_size=None, precision=None, scale=None, is_nullable=True          # noqa: E501
            )
        elif column_type in {"TIMESTAMP", "TIMESTAMP_NS"}:
            return ResultMetadata(
                name=column_name, type_code=8, display_size=None, internal_size=None, precision=0, scale=9, is_nullable=True                # noqa: E501
            )
        else:
            # TODO handle more types
>           raise NotImplementedError(f"for column type {column_type}")
E           NotImplementedError: for column type INTEGER
```

This code worked on 0.3.0

**Comment by tekumara** [comment]
Hmm that's weird, `INTEGER` works when I try your table, but the `BLOB`/`BINARY` and `TIME` types don't, which I'd expect as they are not implemented yet (so I've removed them below):
```python
    cur.execute(
        """
CREATE TABLE IF NOT EXISTS test_types (
    TEST_BIGINT BIGINT,
    TEST_INTEGER INTEGER,
    TEST_SMALLINT SMALLINT,
    TEST_FLOAT FLOAT,
    TEST_FLOAT8 FLOAT8,
    TEST_FLOAT4 FLOAT4,
    TEST_DOUBLE FLOAT,
    TEST_DOUBLE_PRECISION DOUBLE PRECISION,
    TEST_REAL REAL,
    TEST_BOOLEAN BOOLEAN,
    TEST_VARCHAR VARCHAR(100),
    TEST_STRING STRING,
    TEST_TEXT TEXT,
    TEST_CHAR CHAR(10),
    TEST_NVARCHAR NVARCHAR(100),
    TEST_NVARCHAR2 NVARCHAR(100),
    TEST_CHAR_VARYING CHAR VARYING(100),
    TEST_NCHAR_VARYING NCHAR VARYING(100),
    TEST_NCHAR NCHAR(100),
    TEST_CHARACTER CHARACTER(100),
    TEST_DATE DATE,
    TEST_TIMESTAMP TIMESTAMP,
    TEST_DATETIME DATETIME,
    TEST_DECIMAL DECIMAL(10,2),
    TEST_NUMERIC NUMERIC(10,2),
    TEST_NUMBER NUMBER(10,2),
    TEST_TINYINT TINYINT,
    TEST_BYTEINT BYTEINT
);
        """
    )
cur.execute("select * from test_types")
print(cur.description)
```

How are you creating the table?

**Comment by tekumara** [comment]
FYI [v0.5.0]( https://github.com/tekumara/fakesnow/releases/tag/v0.5.0) now supports BINARY and TIME types because I can see you are using them 🙂

**Comment by criccomini** [comment]
Hmm. Let me try again and see if I missed something.

**Comment by criccomini** [comment]
I tested with 0.5.0. The INTEGER error still exists.

This is the test (and line) that fails:

https://github.com/recap-build/recap/blob/main/tests/unit/readers/test_snowflake.py#L76

Full stack trace:

```
tests/unit/readers/test_snowflake.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
recap/readers/dbapi.py:29: in to_recap
    names = [name[0].upper() for name in cursor.description]
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:78: in description
    with self._duck_conn.cursor() as cur:
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:85: in description
    meta = FakeSnowflakeCursor._describe_as_result_metadata(cur.fetchall())  # noqa: SLF001
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:329: in _describe_as_result_metadata
    meta = [
.venv/lib/python3.10/site-packages/fakesnow/fakes.py:330: in <listcomp>
    as_result_metadata(column_name, column_type, null)
```

I also tried dumping the `describe_results` variable in fakes.py and I see it has `INTEGER` types in it:

```
[('table_catalog', 'VARCHAR', 'YES', None, None, None), ('table_schema', 'VARCHAR', 'YES', None, None, None), ('table_name', 'VARCHAR', 'YES', None, None, None), ('column_name', 'VARCHAR', 'YES', None, None, None), ('ordinal_position', 'INTEGER', 'YES', None, None, None), ('column_default', 'VARCHAR', 'YES', None, None, None), ('is_nullable', 'VARCHAR', 'YES', None, None, None), ('data_type', 'VARCHAR', 'YES', None, None, None), ('character_maximum_length', 'INTEGER', 'YES', None, None, None), ('character_octet_length', 'INTEGER', 'YES', None, None, None), ('numeric_precision', 'INTEGER', 'YES', None, None, None), ('numeric_precision_radix', 'INTEGER', 'YES', None, None, None), ('numeric_scale', 'INTEGER', 'YES', None, None, None), ('collation_name', 'INTEGER', 'YES', None, None, None), ('is_identity', 'INTEGER', 'YES', None, None, None), ('identity_generation', 'INTEGER', 'YES', None, None, None), ('identity_cycle', 'INTEGER', 'YES', None, None, None)]
```

I wonder if it's something to do with your `information_schema` impl? Should it be returning BIGINT instead of INTEGER?

**Comment by tekumara** [comment]
Thanks for repro! I hadn't covered the case of describe on the info schema itself.

[v0.5.1](https://github.com/tekumara/fakesnow/releases/tag/v0.5.1) will hopefully fix this for you 🤞

**Comment by criccomini** [comment]
That worked! ❤️ Thanks so much! 😄
